lib/do_auth/router.ex

```diff
@@ -0,0 +1,51 @@
+defmodule DoAuth.Router do
+  @moduledoc """
+  Standard router to provide work with different endpoints.
+  """
+
+  use Plug.Router
+
+  alias Uptight.Binary
+  alias Uptight.Base, as: B
+  alias Uptight.Text, as: T
+  alias Uptight.Result
+  alias :enacl, as: C
+
+  import DynHacks
+
+  import Witchcraft.Functor
+
+  require Logger
+
+  plug(:match)
+
+  plug(Plug.Parsers,
+    parsers: [:json],
+    pass: ["application/json"],
+    json_decoder: Jason
+  )
+
+  plug(:dispatch)
+
+  #Let's define our /doauth/confirm endpoint
+  post "/doauth/confirm/:token" do   # And then we'll define parament for token in path, which we'll use
+
+  #Part where we need to get corresponding token out of credential storage and compare our tokens
+  #TODO: unfortunately, I didn't figured out how to get token from the storage
```

Comment:
I don't think you can. I mean, you can manually scan the storage, credential-by-credential, but it's good that you didn't.

It's because, I think, my vocabulary was a bit off. What I mean with `reserve_identity` [here](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/user.ex#L79) is that we create a credential, which -- really -- should have kind "identity parking", and then it's gonna be respected by the [is email / username reserved or used](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/user.ex#L64) algorithm until it's invalidated (we don't care about invalidation at the moment).

My mistake then is that I do not provide a facility to get a user, together with "parking" credential by the shared secret! To do this, you'll have to:

 - [ ] Find the user by E-Mail in `on_reserve_identity` after you make the "parking" credential and insert a reference to it into its state.
 - [ ] Figure out how to use ETS tables (I always forget the syntax) and make a function just like the [`by_email!`](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/user.ex#L29), except a failable `by_seed`, which returns maybe state of the corresponding user's process.
 - [ ] Use ["amend credential" API](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/credential.ex#L219), which calls [the following GenServer hook](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/credential.ex#L275) to amend the "parked" credential so that it has kind "user activation slip". You can update the reference in the User's state, but it's not really needed! *If* you get your credentials with [`tip`](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/credential.ex#L228), [as you should](https://github.com/doma-engineering/do-auth/issues/16), then the tip shall return the latest credential, which is [reachable](https://github.com/doma-engineering/do-auth/blob/5587c80d9add0a978b3d3c8e0eb4c9ee1a60ad3d/lib/do_auth/credential.ex#L347) by the amendment chain from the source.